From 6eac2b1d091e5ca3e50b8e21747020283659594c Mon Sep 17 00:00:00 2001
From: Yu Chien Peter Lin <peterlin@andestech.com>
Date: Mon, 31 Oct 2022 09:18:57 +0800
Subject: [PATCH] r9a07g043f.dtsi: resize PLMT region

Signed-off-by: Yu Chien Peter Lin <peterlin@andestech.com>
---
 arch/riscv/dts/r9a07g043f.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/riscv/dts/r9a07g043f.dtsi b/arch/riscv/dts/r9a07g043f.dtsi
index 69aaa9bca4..00f369385d 100644
--- a/arch/riscv/dts/r9a07g043f.dtsi
+++ b/arch/riscv/dts/r9a07g043f.dtsi
@@ -106,7 +106,7 @@
 		plmt0@110c0000 {
 			u-boot,dm-spl;
 			compatible = "andestech,plmt0";
-			reg = <0x0 0x110c0000 0x0 0x100000>;
+			reg = <0x0 0x110c0000 0x0 0x10000>;
 			interrupts-extended = <&CPU0_intc 7>;
 		};
 
-- 
2.34.1

