From a90e86ad9d4a6f451a6e8545e363246fd33db743 Mon Sep 17 00:00:00 2001
From: Yu Chien Peter Lin <peterlin@andestech.com>
Date: Wed, 4 Oct 2023 17:02:22 +0800
Subject: [PATCH v4 13/18] riscv_pmu_sbi: fdt-based probe

---
 drivers/perf/riscv_pmu_sbi.c | 25 +++++++++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/drivers/perf/riscv_pmu_sbi.c b/drivers/perf/riscv_pmu_sbi.c
index 4c02c0da0975..aa507e868db6 100644
--- a/drivers/perf/riscv_pmu_sbi.c
+++ b/drivers/perf/riscv_pmu_sbi.c
@@ -770,9 +770,12 @@ static int pmu_sbi_dying_cpu(unsigned int cpu, struct hlist_node *node)
 	return 0;
 }
 
+typedef int (*riscv_pmu_setup_irq_fn)(unsigned int *);
 static int pmu_sbi_setup_irqs(struct riscv_pmu *pmu, struct platform_device *pdev)
 {
 	int ret;
+	struct device *dev = &pdev->dev;
+	riscv_pmu_setup_irq_fn setup_irq_fn;
 	struct cpu_hw_events __percpu *hw_events = pmu->hw_events;
 	struct irq_domain *domain = NULL;
 
@@ -785,10 +788,19 @@ static int pmu_sbi_setup_irqs(struct riscv_pmu *pmu, struct platform_device *pde
 		   riscv_cached_mimpid(0) == 0) {
 		riscv_pmu_irq_num = THEAD_C9XX_RV_IRQ_PMU;
 		riscv_pmu_use_irq = true;
+#if 0
 	} else if (IS_ENABLED(CONFIG_ANDES_CUSTOM_PMU) &&
 		   riscv_cached_mvendorid(0) == ANDES_VENDOR_ID) {
 		riscv_pmu_irq_num = ANDES_SLI_CAUSE_BASE + ANDES_RV_IRQ_PMU;
 		riscv_pmu_use_irq = true;
+#endif
+	}
+
+	setup_irq_fn = of_device_get_match_data(dev);
+	if (setup_irq_fn) {
+		ret = setup_irq_fn(&riscv_pmu_irq_num);
+		if (ret)
+			return ret;
 	}
 
 	if (!riscv_pmu_use_irq)
@@ -949,6 +961,19 @@ static int pmu_sbi_device_probe(struct platform_device *pdev)
 	return ret;
 }
 
+static int andes_setup_irq(unsigned int *irq_num)
+{
+	*irq_num = ANDES_SLI_CAUSE_BASE + ANDES_RV_IRQ_PMU;
+	return 0;
+}
+
+static const struct of_device_id riscv_pmu_of_device_ids[] = {
+	{ .compatible = "riscv,pmu", .data = NULL },
+	{ .compatible = "andestech,pmu", .data = andes_setup_irq },
+	{ },
+};
+MODULE_DEVICE_TABLE(of, riscv_pmu_of_device_ids);
+
 static struct platform_driver pmu_sbi_driver = {
 	.probe		= pmu_sbi_device_probe,
 	.driver		= {
-- 
2.34.1

