From bb9ea564be43b85b135addb07a3fa7af25eed57e Mon Sep 17 00:00:00 2001
From: Yu Chien Peter Lin <peterlin@andestech.com>
Date: Thu, 12 Oct 2023 11:18:10 +0800
Subject: [PATCH v4 13/13] Don't remove pmu node

Upstream-Status: Pending
---
 lib/utils/fdt/fdt_fixup.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/utils/fdt/fdt_fixup.c b/lib/utils/fdt/fdt_fixup.c
index f3a1a8e..a8d89a4 100644
--- a/lib/utils/fdt/fdt_fixup.c
+++ b/lib/utils/fdt/fdt_fixup.c
@@ -487,5 +487,5 @@ void fdt_fixups(void *fdt)
 	fdt_plic_fixup(fdt);
 
 	fdt_reserved_memory_fixup(fdt);
-	fdt_pmu_fixup(fdt);
+	//fdt_pmu_fixup(fdt);
 }
-- 
2.34.1

