From 40d8212f394bb3e716b3a14a32ff72a07d7703ca Mon Sep 17 00:00:00 2001
From: Yu Chien Peter Lin <peterlin@andestech.com>
Date: Wed, 4 Oct 2023 10:57:01 +0800
Subject: [PATCH v4 10/13] TESTING: don't grant write?

Upstream-Status: Pending
---
 platform/generic/andes/andes_pmu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/platform/generic/andes/andes_pmu.c b/platform/generic/andes/andes_pmu.c
index 01b8064..343b562 100644
--- a/platform/generic/andes/andes_pmu.c
+++ b/platform/generic/andes/andes_pmu.c
@@ -111,7 +111,7 @@ int andes_pmu_init(void)
 		return rc;
 
 	/* Enable S-mode write permission */
-	csr_write(CSR_MCOUNTERWEN, 0xfffffffd);
+	//csr_write(CSR_MCOUNTERWEN, 0xfffffffd);
 	/* Inhibit HPM counter in M-mode */
 	csr_write(CSR_MCOUNTERMASK_M, 0xfffffffd);
 	/* Delegate S-mode local interrupt to S-mode */
-- 
2.34.1

